src/cert_store.rs

```diff
@@ -263,6 +268,66 @@ impl<'a> Iterator for Certs<'a> {
     }
 }
 
+/// A builder type for imports of PKCS #12 archives.
+#[derive(Default)]
+pub struct PfxImportOptions {
+    password: Option<Vec<u16>>,
+    flags: winapi::DWORD,
+}
+
+impl PfxImportOptions {
+    /// Returns a new `PfxImportOptions` with default settings.
+    pub fn new() -> PfxImportOptions {
+        PfxImportOptions::default()
+    }
+
+    /// Sets the password to be used to decrypt the archive.
+    pub fn password(&mut self, password: &str) -> &mut PfxImportOptions {
+        self.password = Some(password.encode_utf16().chain(Some(0)).collect());
+        self
+    }
+
+    /// If set, the private key in the archive will not be persisted.
+    ///
+    /// If not set, private keys are persisted on disk and must be manually deleted.
+    pub fn no_persist_key(&mut self, no_persist_key: bool) -> &mut PfxImportOptions {
+        self.flag(PKCS12_NO_PERSIST_KEY, no_persist_key)
+    }
+
+    /// If set, all extended properties of the certificate will be imported.
+    pub fn include_extended_properties(&mut self,
+                                       include_extended_properties: bool)
+                                       -> &mut PfxImportOptions {
+        self.flag(PKCS12_INCLUDE_EXTENDED_PROPERTIES, include_extended_properties)
+    }
+
+    fn flag(&mut self, flag: winapi::DWORD, set: bool) -> &mut PfxImportOptions {
+        if set {
+        self.flags |= flag;
```

Comment:
@sfackler 
Is the indentation intended here and below intended? Looks a bit strange to me.